ESLint-Formatter.py

```diff
@@ -143,11 +143,37 @@ def get_node_path():
     return node
 
   @staticmethod
+  def get_local_eslint(root, folders):
+    pkg = os.path.join(root, 'package.json')
+    esl = 'node_modules/.bin/eslint'
+
+    # find package.json first
+    if os.path.isfile(pkg):
+      # get eslint from local node_modules
+      return os.path.join(root, esl)
+    else:
+      fp = folders.pop()
+      if fp:
+        fd = root.split(fp)[0]
+        if fd:
+          return PluginUtils.get_local_eslint(fd, folders)
+
+    return None
+
+  @staticmethod
   def get_eslint_path():
     platform = sublime.platform()
-    eslint = PluginUtils.get_pref("eslint_path").get(platform)
-    print("Using eslint path on '" + platform + "': " + eslint)
-    return eslint
+    cwd = os.getcwd()
+    folders = cwd.split('/')
+
+    esl = PluginUtils.get_local_eslint(cwd, folders)
```

Comment:
Why does the cwd and an array of the cwd both need to be passed in? Passing both creates two source of truths in `get_local_eslint` which gets awkward when trying to understand the code. Having only one should also make line 157 `fd = root.split(fp)[0]` cleaner.